iso15118/shared/security.py

```diff
@@ -1117,6 +1117,69 @@ def decrypt_priv_key(
     raise DecryptionError()
 
 
+def derive_certificate_hash_data(certificate: bytes) -> Dict[str, str]:
+    """Extract certificate hash data to be used in an OCPP AuthorizeRequest.
+
+    Args:
+        certificate: The certificate in binary (DER) form.
+
+    Returns:
+        A dictionary with all information required for an OCSPRequestDataType
+        (2.36. OCSPRequestDataType, p. 382, OCPP 2.0.1 Part 2)
+
+    Raises:
+        CertAttributeError: if a certificate is provided with a hash algorithm
+            that OCPP doesn't accept.
+            Only SHA256, SHA384, and SHA512 are allowed.
+            (3.42 HashAlgorithmEnumType, p. 403, OCPP 2.0.1 Part 2)
+    """
+    certificate = load_der_x509_certificate(certificate)
+    public_key = certificate.public_key()
+    public_key_bytes = public_key.public_bytes(
+        encoding=Encoding.X962,
+        format=PublicFormat.UncompressedPoint,
+    )
+    # TODO: is this correct?
+    # gives a name like:
+    # 'DC=MO,C=DE,O=Keysight Technologies,CN=PKI-1_CRT_MO_SUB2_VALID'
+    # Do we need this whole thing or only, say, O?
+    distinguished_name = certificate.issuer.rfc4514_string().encode()
```

Comment:
that is correct. The distinguisher name is the issuer:
https://www.ibm.com/docs/en/i/7.2?topic=concepts-distinguished-name
https://www.ibm.com/docs/en/ibm-mq/7.5?topic=certificates-distinguished-names

you can compare it when you extract the issuer, check here the command:
https://www.notion.so/switchev/Certificates-944733c8b468493a9e0eeace3b772d6a

Reply:
sounds good, I am all in favour of comprehensive comments ;)